etl/steps/data/garden/ggdc/2020-10-01/ggdc_maddison_project_database.py

```diff
@@ -0,0 +1,245 @@
+        "Sub-Sahara Africa.1",
+        "World",
+    ]
+    additional_population_data = additional_data[population_columns]
+    additional_population_data.columns = [
+        region.replace(".1", "") for region in additional_population_data.columns
+    ]
+    additional_population_data = additional_population_data.melt(
+        id_vars="Region", var_name="Country", value_name="Population"
+    ).rename(columns={"Region": "Year"})
+
+    # Prepare additional GDP data.
+    gdp_columns = [
+        "Region",
+        "Western Europe",
+        "Eastern Europe",
+        "Western Offshoots",
+        "Latin America",
+        "Asia (East)",
+        "Asia (South and South-East)",
+        "Middle East",
+        "Sub-Sahara Africa",
+        "World GDP pc",
+    ]
+    additional_gdp_data = additional_data[gdp_columns].rename(
+        columns={"World GDP pc": "World"}
+    )
+    additional_gdp_data = additional_gdp_data.melt(
+        id_vars="Region", var_name="Country", value_name=GDP_PER_CAPITA_COLUMN
+    ).rename(columns={"Region": "Year"})
+
+    # Merge additional population and GDP data.
+    additional_combined_data = pd.merge(
+        additional_population_data,
+        additional_gdp_data,
+        on=["Year", "Country"],
+        how="inner",
+    )
+    # Convert units.
+    additional_combined_data["Population"] = (
+        additional_combined_data["Population"] * 1000
+    )
+
+    # Create column for GDP.
+    additional_combined_data[GDP_COLUMN] = (
+        additional_combined_data[GDP_PER_CAPITA_COLUMN]
+        * additional_combined_data["Population"]
+    )
+
+    # Standardize country names.
+    countries = load_countries()
+    additional_combined_data["Country"] = additional_combined_data["Country"].replace(
+        countries
+    )
+
+    assert len(additional_combined_data) == len(additional_population_data)
+    assert len(additional_combined_data) == len(additional_gdp_data)
+
+    return additional_combined_data
+
+
+def generate_ggdc_data(data_file: str) -> pd.DataFrame:
+    """Load and process GGDC data.
+
+    Parameters
+    ----------
+    data_file : str or Path
+        Path to original data file.
+
+    Returns
+    -------
+    combined : pd.DataFrame
+        Processed GGDC data.
+
+    """
+    # Load main and additional GDP data.
+    gdp_data = load_main_data(data_file=data_file)
+    additional_data = load_additional_data(data_file=data_file)
+
+    # Combine both dataframes.
+    combined = (
+        pd.concat([gdp_data, additional_data], ignore_index=True)
+        .dropna(how="all", subset=[GDP_COLUMN, GDP_PER_CAPITA_COLUMN])
+        .sort_values(["Country", "Year"])
+        .reset_index(drop=True)
+    )
+
+    # Some rows have spurious zero GDP. Remove them.
+    zero_gdp_rows = combined[GDP_COLUMN] == 0
+    if zero_gdp_rows.any():
+        print(f"WARNING: Removing {zero_gdp_rows.sum()} rows with zero GDP.")
+        combined = combined[~zero_gdp_rows].reset_index(drop=True)
```

Comment:
A warning suggests that it's unexpected. Better to QA them and explicitly remove them, and only warn if unexpected zeros are there. Otherwise everyone who builds this data will wonder if something's gone wrong.